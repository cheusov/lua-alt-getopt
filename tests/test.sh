#!/bin/sh

runtest (){
    printf '=================================================================\n'
    printf '======= args: %s\n' "$*"
    "$@" 2>&1 | awk '/stack traceback:/, /\[C\]: [?]/ {next} {print}'
}

do_test_getopt (){
    runtest ../alt_getopt -h -
    runtest ../alt_getopt --help
    runtest ../alt_getopt -h --help -v --verbose -V -o 123 -o234
    runtest ../alt_getopt --output 123 --output 234 -n 999 -n9999 --len 5 --fake /dev/null
    runtest ../alt_getopt -hVv -- -1 -2 -3
    runtest ../alt_getopt --fake -v -- -1 -2 -3
    runtest ../alt_getopt - -1 -2 -3
    runtest ../alt_getopt --fake -v - -1 -2 -3
    runtest ../alt_getopt -1 -2 -3
    runtest ../alt_getopt -hvV
    runtest ../alt_getopt -ho 123
    runtest ../alt_getopt -hoV 123
    runtest ../alt_getopt --unknown
    runtest ../alt_getopt --output='file.out' -nNNN --len=LENGTH
    runtest ../alt_getopt --output --file--

    runtest ../alt_getopt --output
    runtest ../alt_getopt -ho
    runtest ../alt_getopt --help -o
    runtest ../alt_getopt --help=value
    runtest ../alt_getopt -ofile1 --set_value 111 --output file2 \
	--set-output=file3

    true
}

do_test (){
    do_test_getopt
}

OBJDIR=${OBJDIR:=.}

do_test > $OBJDIR/_test.res 2>&1

if ! diff -u test.out $OBJDIR/_test.res; then
    echo "rewrite fails" 1>&2
    exit 1
fi

