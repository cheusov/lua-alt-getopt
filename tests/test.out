=================================================================
======= args: ../alt_getopt -h -
option `h'
fin-option `h': 1
ARGV [2] = -
=================================================================
======= args: ../alt_getopt --help
option `h'
fin-option `h': 1
=================================================================
======= args: ../alt_getopt -h --help -v --verbose -V -o 123 -o234
option `h'
option `h'
option `v'
option `v'
option `V'
option `o': 123
option `o': 234
fin-option `V': 1
fin-option `h': 1
fin-option `o': 234
fin-option `v': 1
=================================================================
======= args: ../alt_getopt --output 123 --output 234 -n 999 -n9999 --len 5 --fake /dev/null
option `o': 123
option `o': 234
option `n': 999
option `n': 9999
option `len': 5
option `fake'
fin-option `len': 5
fin-option `fake': 1
fin-option `o': 234
fin-option `n': 9999
ARGV [11] = /dev/null
=================================================================
======= args: ../alt_getopt -hVv -- -1 -2 -3
option `h'
option `V'
option `v'
fin-option `h': 1
fin-option `v': 1
fin-option `V': 1
ARGV [3] = -1
ARGV [4] = -2
ARGV [5] = -3
=================================================================
======= args: ../alt_getopt --fake -v -- -1 -2 -3
option `fake'
option `v'
fin-option `fake': 1
fin-option `v': 1
ARGV [4] = -1
ARGV [5] = -2
ARGV [6] = -3
=================================================================
======= args: ../alt_getopt - -1 -2 -3
ARGV [1] = -
ARGV [2] = -1
ARGV [3] = -2
ARGV [4] = -3
=================================================================
======= args: ../alt_getopt --fake -v - -1 -2 -3
option `fake'
option `v'
fin-option `fake': 1
fin-option `v': 1
ARGV [3] = -
ARGV [4] = -1
ARGV [5] = -2
ARGV [6] = -3
=================================================================
======= args: ../alt_getopt -1 -2 -3
Unknown option `-1'
=================================================================
======= args: ../alt_getopt -hvV
option `h'
option `v'
option `V'
fin-option `h': 1
fin-option `V': 1
fin-option `v': 1
=================================================================
======= args: ../alt_getopt -ho 123
option `h'
option `o': 123
fin-option `o': 123
fin-option `h': 1
=================================================================
======= args: ../alt_getopt -hoV 123
option `h'
option `o': V
fin-option `o': V
fin-option `h': 1
ARGV [2] = 123
=================================================================
======= args: ../alt_getopt --unknown
Unknown option `--unknown'
=================================================================
======= args: ../alt_getopt --output=file.out -nNNN --len=LENGTH
option `o': file.out
option `n': NNN
option `len': LENGTH
fin-option `len': LENGTH
fin-option `o': file.out
fin-option `n': NNN
=================================================================
======= args: ../alt_getopt --output --file--
option `o': --file--
fin-option `o': --file--
=================================================================
======= args: ../alt_getopt --output
Missed value for option `--output'
=================================================================
======= args: ../alt_getopt -ho
Missed value for option `-o'
=================================================================
======= args: ../alt_getopt --help -o
Missed value for option `-o'
=================================================================
======= args: ../alt_getopt --help=value
Bad usage of option `--help=value'
=================================================================
======= args: ../alt_getopt -ofile1 --set_value 111 --output file2 --set-output=file3
option `o': file1
option `S': 111
option `o': file2
option `o': file3
fin-option `o': file3
fin-option `S': 111
